msc_pygeoapi/loader/radar_coverage_realtime.py

```diff
@@ -0,0 +1,487 @@
+
+        # identify expect intervals not retrieved from ES
+        diff = list(set(expected_intervals) - set(retrieved_datetimes))
+        diff.sort()
+        diff_str = [dt.strftime(DATETIME_RFC3339_FMT) for dt in diff]
+
+        # if there are expected intervals missing log an error
+        if diff:
+            LOGGER.error(
+                f'Missing {self.geomet_layer_name} forecast hours (n='
+                f'{len(diff)}) detected between {start} and {forecast_time}: '
+                f'{", ".join(diff_str)}'
+            )
+
+        return retrieved_datetimes == expected_intervals
+
+    def generate_precip_config_dict(self, datetime=None):
+        """
+        Generates a dictionary of default datetime and 3-hour ISO8601 period
+        for a given precipitation type.
+        :param datetime: `datetime` of default forecast time to set
+        :returns: `dict` of precipitation temporal config
+        """
+
+        if datetime is None:
+            datetime = self.datetime
+
+        start = (self.datetime - timedelta(hours=3)).strftime(
+            DATETIME_RFC3339_FMT
+        )
+        end = self.datetime.strftime(DATETIME_RFC3339_FMT)
+
+        config = {
+            'default_time': datetime.strftime(DATETIME_RFC3339_FMT),
+            'time_extent': f'{start}/{end}/PT6M',
+        }
+
+        return config
+
+    def update_temporal_config(self):
+        '''
+        Create/update the temporal configuration file
+        for the precipitation type/coverage type combination
+        returns: `bool` of update status
+        '''
+        updated_config = None
+
+        if self.config_file.exists():
+            with self.config_file.open() as f:
+                config = json.load(f)
+                # get previous default time for precip type
+                previous_default_datetime = datetime.strptime(
+                    config['default_time'],
+                    DATETIME_RFC3339_FMT,
+                )
+                # update temporal config if new default time is
+                # later than previous
+                if self.datetime > previous_default_datetime:
+                    # verify that the previous 31 timesteps are available
+                    self.verify_full_extent_available(self.datetime)
+                    # update config
+                    updated_config = {
+                        **config,
+                        **self.generate_precip_config_dict(),
+                    }
+
+        # temporal config file does not exist
+        else:
+            # check full extent available for current datetime
+            # and create config file if it is
+            if self.verify_full_extent_available(self.datetime):
+                updated_config = self.generate_precip_config_dict()
+
+        if not updated_config:
+            return False
+
+        LOGGER.info(
+            f'Updating radar coverage {self.precip_type.upper()} temporal'
+            f' configuration file...'
```

Comment:
f-string not required here.

Reply:
Since I use a string-formatted literal in the first string, I have no choice but to have the second line continuing as a f-string.